WebApi/BookService/LibraryList.cs

```diff
@@ -0,0 +1,228 @@
+    using System.Linq;
+    using WebApi.Models;
+
+    /// <summary>
+    /// Implementation of IBook
+    /// </summary>
+    public class LibraryList : ILibrary
+    {
+        /// <summary>
+        /// List of books
+        /// </summary>
+        private readonly List<Book> books;
+
+        /// <summary>
+        /// List of authors
+        /// </summary>
+        private readonly List<Author> authors;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LibraryList"/> class
+        /// </summary>
+        public LibraryList()
+        {
+            this.authors = new List<Author>()
+            {
+                new Author("SurnameA", "NameA", "PatronymicA"),
+                new Author("SurnameB", "NameB", "PatronymicB"),
+                new Author("SurnameC", "NameC", "PatronymicC")
+            };
+            this.books = new List<Book>()
+            {
+                new Book("First book", 0),
+                new Book("Second book", 1),
+                new Book("Third book", 2),
+                new Book("Fourth book", 0),
+                new Book("Fifth book", 1),
+                new Book("Sixth book", 2)
+            };
+        }
+
+        /// <summary>
+        /// Adding author
+        /// </summary>
+        /// <param name="author">Author to add</param>
+        /// <returns>Id of author inside collection or db</returns>
+        public int AddAuthor(Author author)
+        {
+            this.authors.Add(author);
+            return author.AuthorID;
+        }
+
+        /// <summary>
+        /// Adding the book to collection
+        /// </summary>
+        /// <param name="book">Book to add</param>
+        /// <returns>Book id</returns>
+        public int AddBook(Book book)
+        {
+            Author bookAuthor = this.authors.FirstOrDefault(author => author.AuthorID == book.AuthorId);
+            if (bookAuthor == null)
+            {
+                book.AuthorId = null;
+            } // null if author doesn`t exist
+
+            this.books.Add(book);
+            return book.Id;
+        }
+
+        /// <summary>
+        /// Getting all authors
+        /// </summary>
+        /// <returns>IEnumerable of authors</returns>
+        public IEnumerable<Author> GetAllAuthors()
+        {
+            return this.authors;
+        }
+
+        /// <summary>
+        /// Getting list of books
+        /// </summary>
+        /// <returns>List of books</returns>
+        public IEnumerable<Book> GetAllBooks()
+        {
+            return this.books;
+        }
+
+        /// <summary>
+        /// Getting author bookss
+        /// </summary>
+        /// <param name="authorId">Author id</param>
+        /// <returns>List of books of author</returns>
+        public IEnumerable<Book> GetAuthorBooks(int authorId)
+        {
+            List<Book> authorBooks = (from book in this.books where book.AuthorId == authorId select book).ToList();
```

Comment:
@MaksimTsedov  @UMSF-K13-1-Samoylenko 
There are two approaches.
First one -  creates some Class DataProvider or DataAccess or Repository and add to it all methods regarding data requests. Then BL layer will call DA later for any data.  This patter is called Repository 
https://docs.microsoft.com/en-us/aspnet/mvc/overview/older-versions/getting-started-with-ef-5-using-mvc-4/implementing-the-repository-and-unit-of-work-patterns-in-an-asp-net-mvc-application

Second, as I mentioned in the comments up, inject in this class DataProvider and use collections directly from it. (almost like current implementation but will be (this.DataProver.Books) 

As for me, first approach is better when you have alot of business logic and to avoid mess of data retrieve and logic it`s nice to create separate DAL layer. 

As for your app. Your business logic is actually data access logic. And your library list is an implementation of the repository pattern.  So if you will create independent DAL your BL will be empty.